policyengine_api/endpoints/search/search.py

```diff
@@ -93,10 +94,21 @@ def get_search(country_id: str, k=10):
     if type not in ["parameter", "variable"]:
         raise ValueError("Type must be parameter or variable")
 
+    if results_param is not None:
+        try:
+            max_results = int(results_param)
+            if max_results <= 0:
+                raise ValueError
+            max_results = min(max_results, 50)  # Limit maximum results to 50
+        except ValueError:
+            return {
```

Comment:
Currently, if a user runs this with an invalid `results` query parameter, the API will return your error message (which is good) but with a status code of 200. Instead, it should really return a failure code - in this case, probably a 400. Could you modify this portion to be similar to the `validate_country` function elsewhere in the API to return the proper status?